File: 0300.longest_increasing_subsequence/solution.go
package main

import "fmt"

func lengthOfLIS(nums []int) int {
	length := len(nums)
	if length < 2 {
		return length
	}
	dp := make([]int, length)

	for i := 0; i < length; i++ {
		dp[i] = 1
	}
	for i := 0; i < length; i++ {
		for j := 0; j < i; j++ {
			if nums[i] > nums[j] {
				dp[i] = max([]int{dp[i], dp[j] + 1})
			}
		}
		fmt.Println(dp)
	}
	return max(dp)
}

func max(nums []int) int {
	value := -1
	for _, n := range nums {
		if value < n {
			value = n
		}
	}
	return value
}

File: 0300.longest_increasing_subsequence/solution.py
class Solution:
    def longest_increasing_subsequence(self, nums):
        length = len(nums)
        if length < 2:
            return length
        dp = [1] * length
        for i in range(length):
            for j in range(0, i):
                if nums[i] > nums[j]:
                    dp[i] = max(dp[i], dp[j] + 1)
        return max(dp)


if __name__ == "__main__":
    print(Solution().longest_increasing_subsequence([10, 9, 2, 5, 3, 7, 101, 18]))
